src/test/java/com/ot/restaurant/entities/IoCContainerTest.java

```diff
@@ -0,0 +1,107 @@
+package com.ot.restaurant.entities;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class IoCContainerTest {
+
+  @Test
+  void whenInstantiatingIoCContainerShouldInstanceTheSame() {
+
+    IoCContainer ioc = IoCContainer.getInstance();
+    IoCContainer ioc2 = IoCContainer.getInstance();
+
+    assertEquals(ioc, ioc2);
+  }
+
+  @Test
+  void whenRegisterValueShouldResolveValueThenAssertSucceeds() throws Exception {
+    IoCContainer ioc = IoCContainer.getInstance();
+
+    String key = "keyEntiti";
+    Object value = new Object();
+
+    ioc.register(key, value);
+
+    Object valueResolvedIoC = ioc.resolve(key);
+
+    assertEquals(value, valueResolvedIoC);
+  }
+
+  @Test
+  public void whenRegisterThrowsDuplicatedExceptionThenAssertionSucceeds() {
+    IoCContainer ioc = IoCContainer.getInstance();
+
+    String key1 = "keyEntiti1";
+    Object value1 = new Object();
+
+    String key2 = "keyEntiti1";
+    Object value2 = new Object();
+
+    //    ioc.register(key1, value1);
+
+    Exception thrown =
+        Assertions.assertThrows(
+            Exception.class,
+            () -> {
+              ioc.register(key1, value1);
+              ioc.register(key2, value2);
+            });
+
+    Assertions.assertEquals("Dependency already exists", thrown.getMessage());
+  }
+
+  @Test
+  void whenRegisterThrowsNullExceptionThenAssertionSucceeds() {
+    IoCContainer ioc = IoCContainer.getInstance();
+
+    Object value1 = new Object();
+    String key1 = "Ramon";
+
+    Exception thrown =
+        Assertions.assertThrows(
+            Exception.class,
+            () -> {
+              ioc.register(key1, value1);
+              String key2 = null;
+              ioc.register(key2, value1);
+            });
+
+    Assertions.assertEquals("Key Cannot be null", thrown.getMessage());
+  }
+
+  @Test
+  void whenResolveThrowsNullpointerExceptionThenAssertionSucceeds() {
+    IoCContainer ioc = IoCContainer.getInstance();
+
+    Exception thrown =
+        Assertions.assertThrows(
+            Exception.class,
+            () -> {
+              String key1 = null;
+              ioc.resolve(key1);
+            });
+
+    Assertions.assertEquals("Value cannot be null", thrown.getMessage());
+  }
+
+  @Test
+  void whenResolveThrowsNotFoundExceptionThenAssertionSucceeds() {
+    IoCContainer ioc = IoCContainer.getInstance();
+
+    Object value1 = new Object();
+
+    Exception thrown =
+        Assertions.assertThrows(
+            Exception.class,
```

Comment:
I think you should use the exception class you created

Reply:
done